shared/src/main/scala-2.10/scalajson.ast/package.scala

```diff
@@ -343,4 +345,280 @@ package object ast {
+                var i2: Int = 0
+                while (i2 < resultNegativeEFlag) {
+                  result = result / radix
+                  i2 += 1
+                }
+              } else
+                return None
+            }
+          }
+
+        val maxLenCheck = digitLength <= maxLengthConstant
+
+        if (result < multmin && maxLenCheck && !eFlag)
+          return None
+
+        if (!(digit == 0 && (decimalFlag || eFlag))) {
+
+          if (!negativeEFlag) {
+            result *= radix
+            if (result < limit + digit && maxLenCheck)
+              return None
+            result -= digit
+          }
+        }
+      }
+      i += 1
+    }
+    if (result < limit)
+      None
+    else {
+      if (negative)
+        Some(result.toLong)
+      else
+        Some(-result.toLong)
+    }
+  }
+
+  private[ast] def toBigInt(value: String): Option[BigInt] = {
+    var decimalFlag = false
+    var result: BigInt = 0
+    var negative = false
+    var char: Char = 0
+    var i = 0
+    var eFlag = false
+    var trailingZeroes: BigInt = 0
+    var negativeEFlag: Boolean = false
+    var resultNegativeEFlag: BigInt = 0
+    var digitLength: BigInt = 0
+    if (value.charAt(0) == '-') {
+      negative = true
+      i += 1
+    }
+
+    while (i < value.length) {
+      char = value.charAt(i)
+      if (char == '.')
+        decimalFlag = true
+      else if ((char | 0x20) == 'e') {
+        eFlag = true
+        val charNext = value.charAt(i + 1)
+        if (charNext == '-')
+          negativeEFlag = true
+        if (negativeEFlag || charNext == '+') {
+          i += 1
+        }
+      } else {
+        if (!(eFlag || decimalFlag))
+          digitLength += 1
+
+        val digit = Character.digit(char, radix)
+
+        if (digit == 0)
+          if (!decimalFlag)
+            trailingZeroes += 1
+          else if (!decimalFlag)
+            trailingZeroes = 0
+
+        if (digit != 0 && (decimalFlag || eFlag))
+          if (!negativeEFlag)
+            return None
+          else {
+            if (trailingZeroes != 0) {
+              resultNegativeEFlag *= radix
+              resultNegativeEFlag += digit
+              if (trailingZeroes >= resultNegativeEFlag) {
+                var i2: Int = 0
+                while (i2 < resultNegativeEFlag) {
+                  result = result / radix
+                  i2 += 1
+                }
+              } else
+                return None
+            }
+          }
+
+        if (!(digit == 0 && (decimalFlag || eFlag))) {
+
+          if (!negativeEFlag) {
+            result *= radix
+            result -= digit
+          }
+        }
+      }
+      i += 1
+    }
+    if (negative)
+      Some(result)
+    else
+      Some(-result)
+  }
+
+  private[ast] def toDouble(value: String): Option[Double] = {
+    try {
+      val asDouble = value.toDouble
+      if (BigDecimal(value, MathContext.UNLIMITED) == BigDecimal(asDouble, MathContext.UNLIMITED))
```

Comment:
This isn't helpful functionality. Not only is it slow, it also isn't what one typically wants.  If there is underflow or overflow, that is a problem; otherwise, the Double is only an approximation to the decimal value anyway, so you already know not to rely on it.  Furthermore, I doubt that letting BigDecimal do the comparison is as fast as converting the double back to a string and letting the string comparison function have a go at it.
There are really three cases here:
1. Number is too big to represent.  Return `None`.
2. Number is not zero but too close to zero to represent.  Return `None`.
3. Number can be represented, but possibly inexactly. Return `Some(value.toDouble)`